### PR TITLE
Adding hosted helm chart with deployment script.

### PR DESCRIPTION
**Pull request by HoustonPutman**
This is a PR to host a helm repo in the github pages of this github repository (bloomberg.github.io/solr-operator/charts). 

I still need to fix the documentation in the getting started, and add documentation around doing releases. But I think the logic should be there.

### CODE REVIEW

**Comment by HoustonPutman** [comment]
Tagging @kcmartin & @pawalt for reviews!

**Comment by sepulworld** [comment]
This is awesome! We will definitely use the helm repo here!

File: Makefile [hunk 1]
```diff
@@ -32,6 +32,9 @@ clean:
 mod-tidy:
 	export GO111MODULE=on; go mod tidy
 
+release:
```
Comment (pawalt) [inline Makefile:35]: Maybe `helm-release` or `helm-publish` or something like that? Just don't want this to be confused with actually cutting a new release.

Comment (HoustonPutman) [reply Makefile]: Added some docs around creating releases to make sure I (and others) don't miss anything in the future. I want this task to auto-create the release at some point...

Actually maybe I'll script out the necessary git commands.

Comment (pawalt) [reply Makefile]: Gotcha that totally makes sense. Releases would probably be pretty easy to script out with the github cli, but that's a different PR.

**Pull request merged**